### PR TITLE
organizing tests, add test for settings

### PR DESCRIPTION
**Pull request by saeid3**
This PR adds a test to verify that added settings in af02b86d4424d445bec2bfefe941fec90299f14a works as expected

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -38,5 +39,19 @@ require('../glpk.js').then(glpkjs => {
 		})
 		t.end();
 	});
+  
+  tape('The time limit should kill the solver before finding optimal solution', { timeout: 99999 }, t => {
```
Comment (rregue) [inline test/test.js:43]: Curious: why the large number here for timeout?

File: test/test.js [hunk 2]
```diff
@@ -38,5 +39,19 @@ require('../glpk.js').then(glpkjs => {
...
+    const problem = 'mip2'
+    let json = JSON.parse(fs.readFileSync(`${__dirname}/data/${problem}.json`).toString());
+    const settings = _.defaults({tmLim: 1}, json.settings)
```
Comment (rregue) [inline test/test.js:46]: should we find a way to test the mipGap also?

**Review by rregue — COMMENTED** [review]
I just left a couple of comments around the large value of timeout and if we should also find a way to test the mipGap setting.

File: test/test.js (hunk 1 above)
Comment (saeid3) [reply test/test.js:43]: no reason, I just copy-pasted the other test (in original repo) and I didn't touch it

File: test/test.js (hunk 2 above)
Comment (saeid3) [reply test/test.js:46]: For examining `mipGap` we should have a complicated (big) problem. I don't know if it's a good idea to have it here in the public repo (given that we need to generate a big hard problem).
probably it's better to test it while implementing it in the matcher along with other parameters.

**Review by rregue — APPROVED** [review]
👍

**Pull request merged**